src/etl/gwas_ingest/pics.py

```diff
@@ -0,0 +1,464 @@
+        # Splitting sample descriptions further:
+        .withColumn(
+            "ancestries", f.split(f.col("discoverySample.ancestry"), r",\s(?![^()]*\))")
+        )
+        # Dividing sample sizes assuming even distribution
+        .withColumn(
+            "adjustedSampleSize",
+            f.col("discoverySample.sampleSize") / f.size(f.col("ancestries")),
+        )
+        # Exploding ancestries
+        .withColumn("gwas_catalog_ancestry", f.explode(f.col("ancestries")))
+        # map gwas population to gnomad superpopulation
+        .join(gwascat_2_gnomad_pop, on="gwas_catalog_ancestry", how="left")
+        # Group by sutdies and aggregate for major population:
+        .groupBy("studyId", "gnomadPopulation")
+        .agg(f.sum(f.col("adjustedSampleSize")).alias("sampleSize"))
+        # Calculate proportions for each study
+        .withColumn(
+            "relativeSampleSize",
+            f.col("sampleSize") / f.sum("sampleSize").over(w_study),
+        )
+        .drop("sampleSize")
+    )
+
+
+def _weighted_r_overall(
+    chromosome: Column,
+    study_id: Column,
+    variant_id: Column,
+    tag_variant_id: Column,
+    relative_sample_size: Column,
+    r: Column,
+) -> Column:
+    """Aggregation of weighted R information using ancestry proportions.
+
+    Args:
+        chromosome (Column): Chromosome
+        study_id (Column): Study identifier
+        variant_id (Column): Variant identifier
+        tag_variant_id (Column): Tag variant identifier
+        relative_sample_size (Column): Relative sample size
+        r (Column): Correlation
+
+    Returns:
+        Column: Estimates weighted R information
+    """
+    pseudo_r = f.when(r >= 1, 0.9999995).otherwise(r)
+    zscore_overall = f.sum(f.atan(pseudo_r) * relative_sample_size).over(
+        Window.partitionBy(chromosome, study_id, variant_id, tag_variant_id)
+    )
+    return f.round(f.tan(zscore_overall), 6)
+
+
+def _is_in_credset(
+    chromosome: Column,
+    study_id: Column,
+    variant_id: Column,
+    pics_postprob: Column,
+    credset_probability: float,
+) -> Column:
+    """Check whether a variant is in the XX% credible set.
+
+    Args:
+        chromosome (Column): Chromosome column
+        study_id (Column): Study ID column
+        variant_id (Column): Variant ID column
+        pics_postprob (Column): PICS posterior probability column
+        credset_probability (float): Credible set probability
+
+    Returns:
+        Column: Whether the variant is in the credible set
+    """
+    w_cumlead = (
+        Window.partitionBy(chromosome, study_id, variant_id)
+        .orderBy(f.desc(pics_postprob))
+        .rowsBetween(Window.unboundedPreceding, Window.currentRow)
+    )
+    pics_postprob_cumsum = f.sum(pics_postprob).over(w_cumlead)
+    w_credset = Window.partitionBy(chromosome, study_id, variant_id).orderBy(
+        pics_postprob_cumsum
+    )
+    return (
+        f.when(f.isnan(pics_postprob), False)
```

Comment:
In what circumstance are the posteriors null?

Reply:
Basically this condition:
```
f.lag(pics_postprob_cumsum, 1).over(w_credset) >= credset_probability, False`
```
was not capturing all negative cases. When `pics_postprob` was null, the credset fell into `True`. It did so, when `_norm_sf(pics_mu, pics_std, neglog_p)` returns np.nan. Mostly, when a variant is not resolved in one of the populations, and the overall R is none.

Reply:
Got it, ty so much for the response.